.github/workflows/ci-full.yaml

```diff
@@ -0,0 +1,78 @@
+name: Full CI Pipeline
+
+on: workflow_dispatch
```

Comment:
Maybe we could also let the full CI pipeline only run for changes on the main branch instead? This way it does not slow down progress in PRs but still uncovers OS-specific errors.

Reply:
At the moment the *full* test is not run automatically. Hence one needs to trigger it manually. When doing so once can usually select the branch it should run this action.. this is at least how I have been seeing this working.

Do you mean something else ?
Do you mean we should use `on: [push]` only and remove `pull_request` ?

Reply:
> Do you mean something else ?
> Do you mean we should use on: [push] only and remove pull_request ?

Exactly, that is what I meant :)

Reply:
Makes sense!
@relu91 any thoughts?